Expand description, add example

natebosch opened the pull request:
Improves the pub score.

Other cleanup:
- Remove optional `new` in example code in the README.
- Use `^` constraints where feasible.

jakemac53 commented on example/client.dart:
```diff
@@ -0,0 +1,16 @@
+  var scratchSpace = await Directory.systemTemp.createTemp();
+  var driver = BazelWorkerDriver(
+      () => Process.start(
+          p.join(sdkDir, 'bin', 'dart'), [p.join('example', 'worker.dart')],
```
nit: 
```suggestion
          Platform.resolvedExecutable, [p.join('example', 'worker.dart')],
```

natebosch replied:
Done.

jakemac53 commented on example/client.dart:
```diff
@@ -0,0 +1,16 @@
+import 'package:bazel_worker/driver.dart';
+import 'package:path/path.dart' as p;
+
+final sdkDir = p.dirname(p.dirname(Platform.resolvedExecutable));
+
+void main() async {
```
```suggestion
void main() async {
```

natebosch replied:
Done.

jakemac53 reviewed (approved):
The example dir could probably use a readme that tells you how to run it?

natebosch commented:
> The example dir could probably use a readme that tells you how to run it?

I added one. I also fleshed out the example a bit more and dropped the usage of `path`.
